games-emulation/mame/mame-0.259.recipe

```diff
@@ -0,0 +1,231 @@
+\"vintage\" software from being lost and forgotten. This is achieved by \
+documenting the hardware and how it functions. The source code to MAME \
+serves as this documentation. The fact that the software is usable serves \
+primarily to validate the accuracy of the documentation (how else can you \
+prove that you have recreated the hardware faithfully?). Over time, MAME \
+(originally stood for Multiple Arcade Machine Emulator) absorbed the \
+sister-project MESS (Multi Emulator Super System), so MAME now documents a \
+wide variety of (mostly vintage) computers, video game consoles and \
+calculators, in addition to the arcade video games that were its initial focus."
+HOMEPAGE="https://www.mamedev.org/"
+COPYRIGHT="1997-2023 MAMEDev and contributors"
+LICENSE="GNU GPL v2"
+REVISION="3"
+SOURCE_URI="https://github.com/mamedev/mame/archive/mame${portVersion/./}.tar.gz"
+CHECKSUM_SHA256="46baf431079a3373ffe8d984b3ba5d62ad5b1d5e356d1f60cf60f6ad03d4cec6"
+SOURCE_FILENAME="mame-$portVersion.tar.gz"
+SOURCE_DIR="mame-mame${portVersion/./}"
+PATCHES="mame-$portVersion.patchset"
+ADDITIONAL_FILES="
+	mame.sh
+	mame.rdef.in"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+bits=64
+optimize=3
+makejobs=$jobArgs
+commandBinDir=$binDir
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	bits=
+	commandBinDir=$prefix/bin
+	commandSuffix=
+	makejobs=1
+	optimize=1
+fi
+
+PROVIDES="
+	mame$secondaryArchSuffix = $portVersion
+	cmd:castool$commandSuffix
+	cmd:chdman$commandSuffix
+	cmd:floptool$commandSuffix
+	cmd:imgtool$commandSuffix
+	cmd:jedutil$commandSuffix
+	cmd:ldresample$commandSuffix
+	cmd:ldverify$commandSuffix
+	cmd:mame$commandSuffix
+	cmd:nltool$commandSuffix
+	cmd:nlwav$commandSuffix
+	cmd:pngcmp$commandSuffix
+	cmd:regrep$commandSuffix
+	cmd:romcmp$commandSuffix
+	cmd:split$commandSuffix
+	cmd:src2html$commandSuffix
+	cmd:srcclean$commandSuffix
+	cmd:testkeys$commandSuffix
+	cmd:unidasm$commandSuffix
+	"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libflac$secondaryArchSuffix
+	lib:libfontconfig$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:liblua$secondaryArchSuffix >= 5.3
+	lib:libpugixml$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libSDL2_2.0$secondaryArchSuffix
+	lib:libSDL2_ttf_2.0$secondaryArchSuffix
+	lib:libsqlite3$secondaryArchSuffix
+	lib:libutf8proc$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:glm
+	devel:libexpat$secondaryArchSuffix
+	devel:libflac$secondaryArchSuffix
+	devel:libfontconfig$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libGL$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:liblua$secondaryArchSuffix >= 5.3
```

Comment:
If you didn't change this localy it will call for lua5.3, this should be changed to:
```diff
-	devel:liblua$secondaryArchSuffix >= 5.3
+	devel:liblua$secondaryArchSuffix >= 5.4
```
Same for lib:liblua*

Reply:
Yep, I had already done that under `REQUIRES` and `BUILD_REQUIRES`.

Is there any chance this has something to do with it?

> MAME now requires Lua compiled as C++ to work correctly. This prevents the use of Lua libraries from Linux distribution package repositories, as they are compiled as C. (The technical reason for this change is that MAME requires C++ stack frames to be unwound correctly, including destructor calls, when Lua errors are raised from C++ code. Using Lua compiled as C will cause resource leaks).

https://www.emunations.com/updates/MAME

Also mentions it in the release notes for 0.253:
https://github.com/mamedev/mame/releases/download/mame0253/whatsnew_0253.txt